manifest/src/lib.rs

```diff
@@ -0,0 +1,81 @@
+use anyhow::{Context, Result};
+use std::fs;
+use std::path::Path;
+use strum::{Display, EnumIter};
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct Manifest {
+    pub manifest_type: ManifestType,
+    pub number_of_dependencies: usize,
+    pub name: String,
+    pub description: Option<String>,
+    pub version: String,
+}
+
+#[derive(Display, Clone, Copy, PartialEq, Eq, Debug, EnumIter)]
+pub enum ManifestType {
+    Npm,
+    Cargo,
+}
+
+pub fn get_manifests<P: AsRef<Path>>(path: P) -> Result<Vec<Manifest>> {
+    let manifest_paths = fs::read_dir(path)?
+        .filter_map(std::result::Result::ok)
+        .map(|entry| entry.path())
+        .filter(|entry| entry.is_file())
+        .filter_map(|file_path| {
+            let file_name = file_path
+                .file_name()
+                .and_then(|name| name.to_str())
+                .unwrap_or_default();
+            file_name_to_manifest_type(file_name).map(|manifest_type| (file_path, manifest_type))
+        })
+        .map(|(file_path, manifest_type)| match manifest_type {
+            ManifestType::Cargo => parse_cargo_manifest(&file_path),
+            ManifestType::Npm => parse_npm_manifest(&file_path),
+        })
+        .filter_map(std::result::Result::ok)
+        .collect::<Vec<_>>();
+
+    Ok(manifest_paths)
+}
+
+fn parse_cargo_manifest(path: &Path) -> Result<Manifest> {
+    let m = cargo_toml::Manifest::from_path(path)?;
+    let package = m.package.context("Not a package (only a workspace)")?;
+    let description = package.description().map(|v| v.into());
+
+    Ok(Manifest {
+        manifest_type: ManifestType::Cargo,
+        number_of_dependencies: m.dependencies.len(),
+        name: package.name.clone(),
+        description,
+        version: package.version().into(),
+    })
+}
+
+fn parse_npm_manifest(path: &Path) -> Result<Manifest> {
+    let package = npm_package_json::Package::from_path(path)?;
+    Ok(Manifest {
+        manifest_type: ManifestType::Npm,
+        number_of_dependencies: package.dependencies.len(),
+        name: package.name,
+        description: package.description,
+        version: package.version,
+    })
+}
+
+fn file_name_to_manifest_type(filename: &str) -> Option<ManifestType> {
+    match filename {
+        "Cargo.toml" => Some(ManifestType::Cargo),
+        "package.json" => Some(ManifestType::Npm),
+        _ => None,
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    #[test]
+    fn test_get_manifest() {}
```

Comment:
Is this a to do?

Reply:
I've just added some integration tests for the `manifest` crate

[01ec7a0](https://github.com/o2sh/onefetch/pull/851/commits/01ec7a0b75ad67ed85ba52d1bd525194f6debc01)
[55b0228](https://github.com/o2sh/onefetch/pull/851/commits/55b022877609c3dd3a1022ffb2dbf288372f39c3)